src/handlers/publish_request_handler.py

```diff
@@ -27,19 +33,36 @@ def handler(event, context):
         return failure_res("No 'body' passed with request.")
     data = json.loads(body)
 
+    # Validate joplin_appname
+    joplin = data.get("joplin_appname")
+    if not joplin:
+        return failure_res("joplin_appname is required")
+
     # Validate janis_branch
     janis_branch = data.get("janis_branch")
     if not janis_branch:
         return failure_res("janis_branch is required")
+    if janis_branch == staging_janis_branch:
+        if not is_staging():
+            return failure_res("Can only deploy to staging Janis from 'staging' Publisher.")
```

Comment:
this prevents us from using a review joplin to accidentally publish to janis staging? great!!

Reply:
yes!